File: test-suite-kotlin/src/test/kotlin/io/micronaut/security/docs/sensitiveendpointrule/SensitiveEndpointRuleReplacement.kt
package io.micronaut.security.docs.sensitiveendpointrule

//tag::imports[]
import io.micronaut.context.annotation.Replaces
import io.micronaut.context.annotation.Requires
import io.micronaut.http.HttpRequest
import io.micronaut.inject.ExecutableMethod
import io.micronaut.management.endpoint.EndpointSensitivityProcessor
import io.micronaut.security.authentication.Authentication
import io.micronaut.security.rules.SecurityRuleResult
import io.micronaut.security.rules.SensitiveEndpointRule
import io.micronaut.security.token.RolesFinder
import jakarta.inject.Singleton
import org.reactivestreams.Publisher
import reactor.core.publisher.Mono
//end::imports[]

@Requires(property = "spec.name", value = "SensitiveEndpointRuleReplacementTest")
//tag::clazz[]
@Replaces(SensitiveEndpointRule::class)
@Singleton
class SensitiveEndpointRuleReplacement(endpointSensitivityProcessor: EndpointSensitivityProcessor) : SensitiveEndpointRule(endpointSensitivityProcessor) {
    override fun checkSensitiveAuthenticated(
        request: HttpRequest<*>,
        authentication: Authentication,
        method: ExecutableMethod<*, *>
    ): Publisher<SecurityRuleResult> = Mono.just(SecurityRuleResult.ALLOWED)
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/security/docs/sensitiveendpointrule/SensitiveEndpointRuleReplacement.py
# tag::imports[]
from jakarta.inject import Singleton
from micronaut.context.annotation import Replaces, Requires
from micronaut.http import HttpRequest
from micronaut.inject import ExecutableMethod
from micronaut.management.endpoint import EndpointSensitivityProcessor
from micronaut.security.authentication import Authentication
from micronaut.security.rules import SecurityRuleResult, SensitiveEndpointRule
from org.reactivestreams import Publisher
from reactor.core.publisher import Mono
# end::imports[]


@Requires(property="spec.name", value="SensitiveEndpointRuleReplacementTest")
# tag::clazz[]
@Singleton
@Replaces(SensitiveEndpointRule)
class SensitiveEndpointRuleReplacement(SensitiveEndpointRule):
    def __init__(self, endpointSensitivityProcessor: EndpointSensitivityProcessor):
        super().__init__(endpointSensitivityProcessor)

    def checkSensitiveAuthenticated(self, request: HttpRequest, authentication: Authentication, method: ExecutableMethod) -> Publisher[SecurityRuleResult]:
        return Mono.just(SecurityRuleResult.ALLOWED)
# end::clazz[]
